### PR TITLE
feat:  Log the validity of the keyringController in the top 3 Migrations that appear in Sentry

### PR DESCRIPTION
**Pull request by MarioAslau**
…migrations that have been reporting errors

## **Description**

As part of the Vault Corruption Tactics: https://www.notion.so/metamask-consensys/Vault-Corruption-98cdfc486e75419fa3e53b7bfe244060

We have to increase our logging around possible areas that could be leading to Vault Corruption, to gather more information.

Add Sentry logs to log the validity of the keyringController in the top 3 Migrations that appear in Sentry, with the highest number of issues.

Issue: https://github.com/MetaMask/mobile-planning/issues/1712

Fixes: vault check in EngineService

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/store/migrations/035.ts [hunk 1]
```diff
@@ -38,6 +38,14 @@ export default async function migrate(stateAsync: unknown) {
     return state;
   }
 
+  const keyringControllerState = state.engine.backgroundState.KeyringController;
+  // @ts-expect-error We are not returning state not to stop the flow of Vault recovery
```
Comment (tommasini) [inline app/store/migrations/035.ts:42]: This comment seems informative, why the ts-expect-error?

File: app/store/migrations/036.test.ts [hunk 2]
```diff
@@ -159,7 +162,10 @@ describe('Migration #036', () => {
         },
         selectedAddress: MOCK_ADDRESS,
       });
+      console.log('OLD STATE:', oldState)
```
Comment (tommasini) [inline app/store/migrations/036.test.ts:165]: Could you remove the Console.log ?

File: app/store/migrations/036.test.ts [hunk 3]
```diff
@@ -159,7 +162,10 @@ describe('Migration #036', () => {
...
+      // why does this always have preferencesController
       const newState = await migrate(oldState);
+      console.log(newState)
```
Comment (tommasini) [inline app/store/migrations/036.test.ts:168]: Here as well?

File: app/store/migrations/036.ts [hunk 4]
```diff
@@ -17,6 +17,7 @@ export interface Identity {
 
 export default async function migrate(stateAsync: unknown) {
   const state = await stateAsync;
+  console.log("STATE:", state)
```
Comment (tommasini) [inline app/store/migrations/036.ts:20]: Here as well

File: app/store/migrations/036.ts [hunk 5]
```diff
@@ -42,6 +43,18 @@ export default async function migrate(stateAsync: unknown) {
     );
     return state;
   }
+
+  const keyringControllerState = state.engine.backgroundState.KeyringController;
+  // @ts-expect-error We are not returning state not to stop the flow of Vault recovery
```
Comment (tommasini) [inline app/store/migrations/036.ts:48]: Hmm what's exactly the typescript error here?

File: app/store/migrations/036.ts [hunk 6]
```diff
@@ -42,6 +43,18 @@ export default async function migrate(stateAsync: unknown) {
...
+  if (!keyringControllerState.vault) {
+    captureException(
+      new Error(
+        // @ts-expect-error We are not returning state not to stop the flow of Vault recovery
```
Comment (tommasini) [inline app/store/migrations/036.ts:52]: And here? What's exactly the typescript error?

File: app/store/migrations/037.ts [hunk 7]
```diff
@@ -29,6 +29,18 @@ export default async function migrate(stateAsync: unknown) {
     );
     return state;
   }
+
+  const keyringControllerState = state.engine.backgroundState.KeyringController;
+  // @ts-expect-error We are not returning state not to stop the flow of Vault recovery
```
Comment (tommasini) [inline app/store/migrations/037.ts:34]: Here as well probably it's not need the TS expect error

**Review by tommasini — CHANGES_REQUESTED** [review]

File: app/store/migrations/037.ts (hunk 7 above)
Comment (MarioAslau) [reply app/store/migrations/037.ts:34]: Screenshot 2024-06-10 at 15 04 37
it's the same everywhere

File: app/store/migrations/036.test.ts [hunk 8]
```diff
@@ -117,6 +118,7 @@ describe('Migration #036', () => {
         engine: {
           backgroundState: {
             PreferencesController: undefined,
+            KeyringController: { vault: {} },
```
Comment (tommasini) [inline app/store/migrations/036.test.ts:121]: On the version before this was introduced we had [Keyring contorller v8](https://github.com/MetaMask/metamask-mobile/blob/04fa71b1730497adfb288ceea655bc07633eed82/package.json#L153)

And at that version the [Keyring Controller already was extending from base controller v2](https://github.com/MetaMask/core/blob/01fb17da36e8a7b9b27f7cff4c200eb6bd6b004d/packages/keyring-controller/src/KeyringController.ts#L240),
Does that mean that the data will be inside of a metadata property?

Comment (tommasini) [reply app/store/migrations/036.test.ts:121]: Hmm this doesn't seem to be truth, we save on the storage without the metadata property will pass this PR

**Review by tommasini — APPROVED** [review]

**Pull request merged**